### PR TITLE
Remove FormFactorRatioAdapter and fix options

### PR DESCRIPTION
**Pull request by mreboud**
Could close issue #369

It also fixes the options for expression observables. Due to a bug, qualified name options were not used.

### CODE REVIEW

**Comment by mreboud** [comment]
(I will fix the merge conflict once the pending PR is on master)

File: eos/form-factors/observables.cc [hunk 1]
```diff
@@ -422,16 +387,17 @@ namespace eos
                         &AnalyticFormFactorBToPLCSR<lcsr::BsToDs>::normalized_moment_1_f_t,
                         std::make_tuple("q2")),
 
-                make_form_factor_adapter("B(_s)->D(_s)::f_0(q2_num)/f_0(q2_denom)",
-                        "B_s->D_s", &FormFactors<PToP>::f_0,
-                        std::make_tuple("q2_num"),
-                        "B->D",     &FormFactors<PToP>::f_0,
-                        std::make_tuple("q2_denom")),
+                make_expression_observable("B(_s)->D(_s)::f_0(q2_num)/f_0(q2_denom)", R"(f_0(q^2_\mathrm{num})/f_+(q^2_\mathrm{denum}))",
```
Comment (dvandyk) [inline eos/form-factors/observables.cc:390]: Typo, denum -> denom, also below.

File: eos/form-factors/observables.cc [hunk 2]
```diff
@@ -422,16 +387,17 @@ namespace eos
...
+                        Unit::None(),
+                        R"(
+                        <<B_s->D_s::f_0(q2)>>[q2=>q2_num]
+                        /
+                        <<B->D::f_0(q2)>>[q2=>q2_denum]
```
Comment (dvandyk) [inline eos/form-factors/observables.cc:395]: Ditto.

File: eos/form-factors/observables.cc [hunk 3]
```diff
@@ -597,25 +559,24 @@ namespace eos
                 make_form_factor_adapter("B->K^*::F_long_T(q2)",
                         &FormFactors<PToV>::f_long_T, std::make_tuple("q2")),
```
Comment (dvandyk) [inline eos/form-factors/observables.cc:561]: This is not removed, maybe this is a display issue or you missed one?

File: eos/form-factors/observables.cc [hunk 4]
```diff
@@ -597,25 +559,24 @@ namespace eos
...
 
-                make_form_factor_adapter("B->K^*::V(q2)/A_1(q2)",
-                        "B->K^*", &FormFactors<PToV>::v,
-                        "B->K^*", &FormFactors<PToV>::a_1,
-                        std::make_tuple("q2")),
+                make_form_factor_adapter("B->K^*::F_long_T(q2)_Normalized",
```
Comment (dvandyk) [inline eos/form-factors/observables.cc:562]: And here you added one?

**Review by dvandyk — CHANGES_REQUESTED** [review]
Only a few oddities, see the comments.

File: eos/form-factors/observables.cc (hunk 3 above)
Comment (mreboud) [reply eos/form-factors/observables.cc:561]: This is ok, what I removed is the ratio, not the adapter.

**Pull request merged**